src/main/java/com/metamx/datatypes/openrtbnative/AssetRequest.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Rad-tech-datatypes.
+ * Copyright 2014 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.datatypes.openrtbnative;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.datatypes.openrtb.Ext;
+import com.metamx.datatypes.openrtb.Video;
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class AssetRequest
+{
+  private final int id;
+  private final int req;
+  private final TitleRequest title;
+  private final ImageRequest img;
+  private final VideoRequest video;
+  private final DataRequest data;
+  private final Ext ext;
+
+  public AssetRequest(
+      @JsonProperty("id") int id,
+      @JsonProperty("req") int req,
+      @JsonProperty("title") TitleRequest title,
+      @JsonProperty("img") ImageRequest img,
+      @JsonProperty("video") VideoRequest video,
+      @JsonProperty("data") DataRequest data,
+      @JsonProperty("ext") Ext ext
+  )
+  {
+    this.id = id;
+    this.req = req;
+    this.title = title;
+    this.img = img;
+    this.video = video;
+    this.data = data;
+    this.ext = ext;
+  }
+
+  @JsonProperty
+  public int getId()
+  {
+    return id;
+  }
+
+  @JsonProperty
+  public int getReq()
```

Comment:
This needs to be called "required", per openRTB. 

you could change the wrapper to say "@JsonProperty("required")", which would override the use of the parameter name, but I think it's cleaner to just call it "required" everywhere.